Tools/cases_generator/analyzer.py

```diff
@@ -1215,6 +1215,77 @@ def analyze_forest(forest: list[parser.AstNode]) -> Analysis:
         instructions, uops, families, pseudos, labels, opmap, first_arg, min_instrumented
     )
 
+#Simple heuristic for size to avoid too much stencil duplication
+def is_large(uop: Uop) -> bool:
+    return len(list(uop.body.tokens())) > 80
+
+def get_uop_cache_depths(uop: Uop) -> Iterator[tuple[int, int, int]]:
+    if uop.name == "_SPILL_OR_RELOAD":
+        for inputs in range(4):
+            for outputs in range(4):
+                if inputs != outputs:
+                    yield inputs, outputs, inputs
+        return
+    if uop.name == "_EXIT_TRACE":
+        for i in range(4):
+            yield i, 0, i
+        return
+    if uop.name in ("_START_EXECUTOR", "_JUMP_TO_TOP", "_DEOPT", "_ERROR_POP_N"):
+        yield 0, 0, 0
+        return
+    non_decref_escape = False
+    for call in uop.properties.escaping_calls.values():
+        if "DECREF" in call.call.text or "CLOSE" in call.call.text:
+            continue
+        non_decref_escape = True
+    has_exit = uop.properties.deopts or uop.properties.side_exit
+    ideal_inputs = 0
+    has_array = False
+    for item in reversed(uop.stack.inputs):
+        if item.size:
+            has_array = True
+            break
+        if item.peek and uop.properties.escapes:
+            break
+        ideal_inputs += 1
+    ideal_outputs = 0
+    for item in reversed(uop.stack.outputs):
+        if item.size:
+            has_array = True
+            break
+        if item.peek and uop.properties.escapes:
+            break
+        ideal_outputs += 1
+    if ideal_inputs > 3:
+        ideal_inputs = 3
+    if ideal_outputs > 3:
+        ideal_outputs = 3
```

Comment:
Can you move the value `3` to a global magic number so that we can play around with increasing/decreasing register counts in the future?